core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -3119,6 +3125,37 @@ private static class ArrayConcatImplementor extends AbstractRexCallImplementor {
     }
   }
 
+  /** Implementor for str_to_map. */
+  private static class StringToMapImplementor extends AbstractRexCallImplementor {
+    StringToMapImplementor() {
+      super("str_to_map", NullPolicy.STRICT, false);
+    }
+
+    @Override Expression implementSafe(RexToLixTranslator translator,
+        RexCall call, List<Expression> argValueList) {
+      switch (call.getOperands().size()) {
+      case 1:
+        return Expressions.call(
```

Comment:
nit: I'm still wondering if the return can be extracted out of the switch statement. You notice that in each case it's "return Expressions.call(...)". What do you think?

Reply:
However, personally, I think the difference is not significant, and many do the same 
for example ItemImplementor/LogImplementor/DatetimeArithmeticImplementor/...